Apps/Database/src/Models/UserPreference.cs

```diff
@@ -38,8 +32,14 @@ public class UserPreference
         public string HdId { get; set; } = null!;
 
         /// <summary>
-        /// Gets or sets a value indicating whether the user dismissed the note popover.
+        /// Gets or sets the preference key.
+        /// </summary>
+        [Key]
+        public string Key { get; set; } = null!;
```

Comment:
I'm not sure why we defined the key as the key and in the notion of the database I don't like the name of key (Maybe Preference and value).  I think this table should be hdid/preference as a composite key.

Reply:
Updated